client/shared/src/search/query/tokens.ts

```diff
@@ -371,129 +371,117 @@ export const hasRegexpValue = (field: string): boolean => {
+                    break
+                case PatternKind.Structural:
+                    decorated.push(...mapStructuralMeta(token))
+                    break
+                case PatternKind.Literal:
+                    decorated.push(token)
+                    break
             }
-            default:
-                decorated.push(token)
+            break
+        case 'filter': {
+            decorated.push({
+                type: 'field',
+                range: token.field.range,
+                value: token.field.value,
+            })
+            if (token.value && token.value.type === 'literal' && hasRegexpValue(token.field.value)) {
+                // Highlight fields with regexp values.
+                decorated.push(
+                    ...decorate({
+                        type: 'pattern',
+                        kind: PatternKind.Regexp,
+                        value: token.value.value,
+                        range: token.value.range,
+                    })
+                )
+            } else if (token.value) {
+                decorated.push(token.value)
+            }
+            break
         }
+        default:
+            decorated.push(token)
     }
     return decorated
 }
 
-const fromDecoratedTokens = (tokens: DecoratedToken[]): Monaco.languages.IToken[] => {
-    const monacoTokens: Monaco.languages.IToken[] = []
-    for (const token of tokens) {
-        switch (token.type) {
-            case 'field':
-            case 'whitespace':
-            case 'keyword':
-            case 'comment':
-            case 'openingParen':
-            case 'closingParen':
-                monacoTokens.push({
-                    startIndex: token.range.start,
-                    scopes: token.type,
-                })
-                break
-            case 'regexpMeta':
-            case 'structuralMeta':
-                /** The scopes value is derived from the token type and its kind.
-                 * E.g., regexpMetaDelimited derives from {@link RegexpMeta} and {@link RegexpMetaKind}.
-                 */
-                monacoTokens.push({
-                    startIndex: token.range.start,
-                    scopes: `${token.type}${token.kind}`,
-                })
-                break
-            default:
+const decoratedToMonaco = (token: DecoratedToken): Monaco.languages.IToken => {
+    switch (token.type) {
+        case 'field':
+        case 'whitespace':
+        case 'keyword':
+        case 'comment':
+        case 'openingParen':
+        case 'closingParen':
+            return {
+                startIndex: token.range.start,
+                scopes: token.type,
+            }
+        case 'regexpMeta':
+        case 'structuralMeta':
+            /** The scopes value is derived from the token type and its kind.
+             * E.g., regexpMetaDelimited derives from {@link RegexpMeta} and {@link RegexpMetaKind}.
+             */
```

Comment:
```suggestion
            // The scopes value is derived from the token type and its kind.
            // E.g., regexpMetaDelimited derives from {@link RegexpMeta} and {@link RegexpMetaKind}.
```

Nit: save docblock syntax for docblock usages :)

Reply:
Ohhh OK I see how this is used now.